api/v2/serializers/directory-serializer.js

```diff
@@ -0,0 +1,148 @@
+const appRoot = require('app-root-path');
+const JsonApiSerializer = require('jsonapi-serializer').Serializer;
+const _ = require('lodash');
+
+const { serializerOptions } = appRoot.require('utils/jsonapi');
+const { openapi } = appRoot.require('utils/load-openapi');
+const { paginate } = appRoot.require('utils/paginator');
+const { apiBaseUrl, resourcePathLink, paramsLink } = appRoot.require('utils/uri-builder');
+
+const directoryResourceProp = openapi.definitions.DirectoryResourceObject.properties;
+const directoryResourceType = directoryResourceProp.type.enum[0];
+const directoryResourceKeys = _.keys(directoryResourceProp.attributes.properties);
+const directoryResourcePath = 'directory';
+const directoryResourceUrl = resourcePathLink(apiBaseUrl, directoryResourcePath);
+
+const ldapKeyToResourceKey = attribute => ({
+  givenName: 'firstName',
+  sn: 'lastName',
+  cn: 'fullName',
+  osuPrimaryAffiliation: 'primaryAffiliation',
+  title: 'jobTitle',
+  osuDepartment: 'department',
+  postalAddress: 'departmentMailingAddress',
+  telephoneNumber: 'officePhoneNumber',
+  osuOfficeAddress: 'officeAddress',
+  facsimileTelephoneNumber: 'faxNumber',
+  mail: 'emailAddress',
+  uid: 'username',
+  osuAltPhoneNumber: 'alternatePhoneNumber',
+  osuUID: 'osuUid',
+}[attribute]);
+
+const resourceKeyToLdapKey = attribute => ({
+  firstName: 'givenName',
+  lastName: 'sn',
+  fullName: 'cn',
+  primaryAffiliation: 'osuPrimaryAffiliation',
+  jobTitle: 'title',
+  department: 'osuDepartment',
+  departmentMailingAddress: 'postalAddress',
+  officePhoneNumber: 'telephoneNumber',
+  officeAddress: 'osuOfficeAddress',
+  faxNumber: 'facsimileTelephoneNumber',
+  emailAddress: 'mail',
+  username: 'uid',
+  alternatePhoneNumber: 'osuAltPhoneNumber',
+  osuUid: 'osuUID',
+}[attribute]);
+
+const valueOperations = (key, value) => {
+  switch (key) {
+    case 'telephoneNumber':
+    case 'osuAltPhoneNumber':
+    case 'facsimileTelephoneNumber': {
+      return `+${value}`;
+    }
+    case 'osuPrimaryAffiliation': {
+      return {
+        S: 'Student',
+        E: 'Employee',
+        O: 'Other',
+        R: 'Retiree',
+        U: 'Unknown',
+      }[value];
```

Comment:
Looks like the map here is pretty much the same as what you did in DAO object expect they are reversed. Can you figure a way to maintain only a map in one place?